homework/index.js

```diff
@@ -30,13 +30,122 @@
+    const trRepo = createAndAppend('tr', tBody);
+    createAndAppend('td', trRepo, { text: 'repo:' });
+    const tdRepoLink = createAndAppend('td', trRepo);
+    createAndAppend('a', tdRepoLink, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+    if (repo.description !== null) {
+      const trDescription = createAndAppend('tr', tBody, {});
+      createAndAppend('td', trDescription, { text: 'Description:' });
+      createAndAppend('td', trDescription, { text: repo.description });
+    }
+    const trForks = createAndAppend('tr', tBody);
+    createAndAppend('td', trForks, { text: 'Forks:' });
+    createAndAppend('td', trForks, { text: repo.forks_count });
+    const trUpdate = createAndAppend('tr', tBody);
+    createAndAppend('td', trUpdate, { text: 'Updated:' });
+    createAndAppend('td', trUpdate, {
+      text: new Date(repo.updated_at).toLocaleString(),
+    });
+  }
+
+  function makeUl(repos) {
+    const rightDiv = document.querySelector('.rightDiv');
+    createAndAppend('p', rightDiv, { class: 'contributor-header', text: 'Contributions' });
+    const ul = createAndAppend('ul', rightDiv, { class: 'contributor-list' });
+    repos.forEach(contributor => {
+      const li = createAndAppend('li', ul, {
+        class: 'contributor-item',
+        tabindex: '0',
+        'aria-label': contributor.login,
+      });
+      createAndAppend('img', li, {
+        src: contributor.avatar_url,
+        class: 'contributor-avatar',
+        height: '48',
+        alt: 'profile photo of contributor',
+      });
+      const contributorData = createAndAppend('div', li, { class: 'contributor-data' });
+      createAndAppend('div', contributorData, { text: contributor.login });
+      createAndAppend('div', contributorData, {
+        class: 'contributor-badge',
+        text: contributor.contributions,
+      });
+      li.onclick = () => {
+        window.open(contributor.html_url, '_blank');
+      };
+    });
+  }
+
+  function makeHtmlOfContents(repos, index) {
+    document.querySelector('.leftDiv').innerHTML = '';
+    document.querySelector('.rightDiv').innerHTML = '';
+    makeTable(repos[index]);
+    fetchJSON(repos[index].contributors_url, (err, fetchedData) => {
+      const root = document.getElementById('root');
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
+        makeUl(fetchedData);
+      }
+    });
+  }
+
+  function mainHtmlConstructor(repos) {
+    makeHeader(repos);
+    const root = document.getElementById('root');
+    const container = createAndAppend('div', root, { id: 'container' });
+    createAndAppend('div', container, { class: 'leftDiv whiteframe' });
+    createAndAppend('div', container, { class: 'rightDiv whiteframe' });
+    const selectMenu = document.querySelector('select');
+    makeHtmlOfContents(repos, selectMenu.options[selectMenu.selectedIndex].value);
```

Comment:
I overlooked this in my first review: the index you need can be obtained either by `selectMenu.selectedIndex` or `selectMenu.value`. In this case both methods give the same result. It is not recommended (and unneeded) to obtain the index value by accessing the `options` elements. Please review the documentation for these links:

https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/selectedIndex

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event#%3Cselect%3E_element

Reply:
ok. actually I don't know how it works. I just looked for some samples from google, those were almost same. I used that and it worked. I hope I will read documentation and understand <select>
and I can make changes I think, if you don't advise against

Reply:
I now understood that my problem was sorting. 
when I used "selectMenu.selectedIndex" directly it works but order is different :) 
eg if I select one repository from Dropdown , table and list filled by another repo :)
and I solved that by sorting the data at the beginning . I sow this in Murat's PR. 
I applied same way. actually now it is easier to read and understand and even write :)
I will push the latest code

Reply:
Yes, that is exactly right. First sort the repositories array that you get back from GitHub and then create the `<option>` element from the sorted array. In this way you maintain a one-to-one correspondence between the index of the selected option and the index of the corresponding repository object. There is no need to create a separate `optionsArr` array. Just use the original sorted repositories array throughout your code